common/src/main/java/tachyon/Version.java

```diff
@@ -25,7 +25,7 @@
 
   static {
     TachyonConf tachyonConf = new TachyonConf();
-    VERSION = tachyonConf.get(Constants.TACHYON_VERSION, "UNDEFINED");
+    VERSION = tachyonConf.get(Constants.TACHYON_VERSION);
```

Comment:
Maybe I misunderstood your comment. In short, right now if a key doesn't exist, getters in `TachyonConf`will throw exception (not using `Preconditions` though, which I will fix later)